exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -179,27 +190,29 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, extensionConfig *runti
 	return ctrl.Result{}, nil
 }
 
-// secretToExtensionConfig maps a secret to ExtensionConfigs with the corresponding InjectCAFromSecretAnnotation
-// to reconcile them on updates of the secrets.
-func (r *Reconciler) secretToExtensionConfig(ctx context.Context, secret client.Object) []reconcile.Request {
-	result := []ctrl.Request{}
-
-	extensionConfigs := runtimev1.ExtensionConfigList{}
-	indexKey := secret.GetNamespace() + "/" + secret.GetName()
+// secretToExtensionConfigFunc returns a func which maps a secret to ExtensionConfigs with the corresponding
+// InjectCAFromSecretAnnotation to reconcile them on updates of the secrets.
+func secretToExtensionConfigFunc[T client.Object](ctrlClient client.Client) handler.TypedMapFunc[T] {
```

Comment:
```suggestion
func secretToExtensionConfigFunc[T *metav1.PartialObjectMetadata](ctrlClient client.Client) handler.TypedMapFunc[T] {
```
maybe

Reply:
keeping this as client.Object allows to be more flexible and use the function for any type which is `client.Object`. In the current case wie use it for `*metav1.PartialObjectMetadata` which fulfills the `client.Object` interface.

Note: `client.Object` is used for similar funcs like this in controller-runtime.

Reply:
Fine for me

Reply:
Was just wondering what the point is of using a generic function if we still just use client.Object :D 

But I know that PartialObjectMetadata is not much more specific than client.Object

Reply:
the point is to fulfill the new signatures from controller-runtime when using WatchesRawSource :-)